ludwig/schema/preprocessing.py

```diff
@@ -0,0 +1,751 @@
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in an audio column",
+    )
+
+    in_memory: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Defines whether the audio dataset will reside in memory during the training process or will be "
+        "dynamically fetched from disk (useful for large datasets). In the latter case a training batch "
+        "of input audio will be fetched from disk each training iteration.",
+    )
+
+    padding_value: Optional[float] = schema_utils.NonNegativeFloat(
+        default=0.0, allow_none=False, description="Float value that is used for padding."
+    )
+
+    norm: Optional[str] = schema_utils.StringOptions(
+        ["per_file"],
+        default=None,
+        allow_none=True,
+        description="Normalization strategy for the audio files. If None, no normalization is performed. If "
+        "per_file, z-norm is applied on a 'per file' level",
+    )
+
+    audio_feature: Optional[dict] = schema_utils.Dict(
+        default={
+            "type": "fbank",
+            "window_length_in_s": 0.04,
+            "window_shift_in_s": 0.02,
+            "num_filter_bands": 80,
+        },
+        description="Dictionary that takes as input the audio feature type as well as additional parameters if type "
+        "!= raw. The following parameters can/should be defined in the dictionary ",
+    )
+
+
+@dataclass
+@register_preprocessor("timeseries")
+class TimeseriesPreprocessingConfig(schema_utils.BaseMarshmallowConfig):
+
+    tokenizer: Optional[str] = schema_utils.StringOptions(
+        sorted(list(tokenizer_registry.keys())),
+        default="space",
+        allow_none=False,
+        description="Defines how to map from the raw string content of the dataset column to a sequence of elements.",
+    )
+
+    timeseries_length_limit: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        allow_none=False,
+        description="Defines the maximum length of the timeseries. All timeseries longer than this limit are cut off.",
+    )
+
+    padding_value: Optional[float] = schema_utils.NonNegativeFloat(
+        default=0.0,
+        allow_none=False,
+        description="Float value that is used for padding.",
+    )
+
+    padding: Optional[str] = schema_utils.StringOptions(
+        ["left", "right"],
+        default="right",
+        allow_none=False,
+        description="the direction of the padding. right and left are available options.",
+    )
+
+    missing_value_strategy: Optional[str] = schema_utils.StringOptions(
+        ["fill_with_const", "fill_with_mode", "fill_with_mean", "backfill", "bfill", "pad", "ffill", "drop_row"],
+        default="fill_with_const",
+        allow_none=False,
+        description="What strategy to follow when there's a missing value in a text column",
+    )
+
+    fill_value: Optional[str] = schema_utils.String(
+        default="",
```

Comment:
Editing comment: I'm not sure what a reasonable default is for Time Series. Glad to pair up on possible solutions!

Reply:
I think this is a tricky one to set since a timeseries column could come in many forms. I will defer to the #ludwig channel to see if people have ideas for a better default.